ibis/backends/dask/execution/selection.py

```diff
@@ -0,0 +1,376 @@
+"""Dispatching code for Selection operations.
+"""
+
+from __future__ import absolute_import
+
+import functools
+import operator
+from collections import OrderedDict
+from operator import methodcaller
+from typing import Optional
+
+import dask.dataframe as dd
+import numpy as np
+import pandas
+from multipledispatch import Dispatcher
+from toolz import compose, concat, concatv, first, unique
+
+import ibis.expr.operations as ops
+import ibis.expr.types as ir
+from ibis.backends.pandas.execution import util
+from ibis.backends.pandas.execution.selection import execute, execute_node
+from ibis.expr.scope import Scope
+from ibis.expr.typing import TimeContext
+
+from ..execution import constants
+
+compute_projection = Dispatcher(
+    'compute_projection',
+    doc="""\
+Compute a projection, dispatching on whether we're computing a scalar, column,
+or table expression.
+
+Parameters
+----------
+expr : Union[ir.ScalarExpr, ir.ColumnExpr, ir.TableExpr]
+parent : ops.Selection
+data : dd.DataFrame
+scope : Scope
+timecontext:Optional[TimeContext]
+
+Returns
+-------
+value : scalar, dd.Series, dd.DataFrame
+
+Notes
+-----
+:class:`~ibis.expr.types.ScalarExpr` instances occur when a specific column
+projection is a window operation.
+""",
+)
+
+
+@compute_projection.register(ir.ScalarExpr, ops.Selection, dd.DataFrame)
```

Comment:
so for example this makes sense for a common daskpandas section as this will be the same implementation for a dd.DataFrame and pd.DataFrame; there might be a number of these.